DiscordBot/Program.cs

```diff
@@ -1,27 +1,41 @@
-﻿using System;
-using System.Collections.Generic;
-using DiscordBot.Commands;
-using DiscordBot.Config;
-
-namespace DiscordBot
-{
-    class Program
-    {
-        static Dictionary<string, ICommand> Commands = new Dictionary<string, ICommand>
-        {
-            { "ping", new Ping() }
-        };
-        
-        static void Main(string[] args)
-        {
-            // fuck tom.... <3
-            var configGetter = new JsonConfigGetter($"config.json");
-            var config = configGetter.GetConfig();
-            var commandGetter = new CommandGetter(Commands, config.Prefix);
-
-            var bot = new Bot(commandGetter, config);
-            
-            bot.Run().ConfigureAwait(false).GetAwaiter().GetResult();
-        }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using DiscordBot.Backends;
+using DiscordBot.Backends.CLI;
+using DiscordBot.Backends.Discord;
+using DiscordBot.Config;
+using DSharpPlus.EventArgs;
+using TicTacToe;
+using botCommands = DiscordBot.Commands;
+
+namespace DiscordBot
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // Discord v CLI
+            var settings = new DiscordSettings();
+
+            var ticTacToeCommand =
+                new botCommands.TicTacToeCommand(new TicTacToeManager<char>(), settings.TranslationDict);
+            
+            var commands = new Dictionary<string[], botCommands.ICommand>
+            {
+                { new []{"ping"}, new botCommands.PingCommand()},
+                {new []{"tic", "tac", "toe"}, ticTacToeCommand}
+            };
+            
+            var configGetter = new JsonConfigGetter("config.json");
+            var config = configGetter.GetConfig();
+            var commandProvider = new botCommands.CommandProvider(commands, config.Prefix);
+            
+            var type = typeof(Bot<>).MakeGenericType(settings.BackendType);
+            dynamic bot = Activator.CreateInstance(type, commandProvider, config);

```

Comment:
I don't see a reason for any of this stuff here

Reply:
then where tf should i put it lmao dummy
this is the proper way (at least the most upvoted answer from stack overflow) i found for my problem. That way i only need to change 1 line to switch backends :D